### ISSUE TITLE
not every permission listed in tab autocomplete (essentialsX)

### ISSUE THREAD

**Issue by Blighthorn** [lucko/LuckPerms#702]
I am using EssentialsX 2.0.1 (Build 578) base+antibuild+chat+spawn.
Luckperm tab autocomplete suggests only few permissions:
*, fly, gamemode, mail, motd, sleepingignored, spawn-on-join, speed, teleport

theres a LOT permissions missing here.
On Build 571 it was ok, but I wanted to add antibuild and upgrade. Maybe permission list grows too much?

Once I had all the list, I restarted server and joined game to check several times but cannot reproduce proper behavior.
Now there is only short list all the time.
At server start there are no error messages about essentials or LP.
I am using Vault too.

**Comment by Blighthorn** [comment]
I have downgraded essentialsX to build 488, removing all addons - but still the same... 
Strange, because before it was ok. Seems, like list of permissions is not properly collected by vault or LP. I don't know what to do.

**Comment by Blighthorn** [comment]
Interesting finding, after I use command like /home then in the autocomplete list appear .home and .sethome
I am now puzzled.

**Comment by spannerman79** [comment]
Luckperm tab autocomplete only applies to `lp`/`lpb` commands. Not to essentialsx commands.

Wrong issue tracker.

**Comment by Blighthorn** [comment]
Hey, the issue is not about autocomplete commands, but permission nodes.

**Comment by spannerman79** [comment]
Did you actually read the wiki?
https://github.com/lucko/LuckPerms/wiki/Command-Usage#lp-tree

> The tree is built using data exposed to the server by plugins, and **expanded over time as plugins check for permissions**.

Tell me how is Luckperms able to tell what a perm node is unless it does a check for it? By "magically" guessing - which will be a danger in on itself.  Or by node checks being performed - as outlined in `tree`

**Comment by Blighthorn** [comment]
Ok, thx for answer. I thought that all persmission nodes are known as plugins are enabled at server start, and not exposed in time. That was because I always saw all permission nodes for other plugins, and I am pretty sure sometimes I saw all for essentialsX too. Usually long list is what I git, thus I thought that maybe now there is something wrong, not then, beacause I upgraded LP to newest version not long ago. That's pity, because that autocomplete is one of main reasons I use LuckPerms, to list nodes without looking for them in web and always get correct spelling. If you are interested, here is my [tree](https://gist.github.com/anonymous/ffcc246cb7a137d1643e69daaa1856c3). I also remember to get some autocomplete nodes for some plugins by doble dot (like pluginname..permnode), but not now. So the only exception where I don't have whole list is EssentialsX now.
